src/optimizer.cpp

```diff
@@ -198,25 +199,22 @@ void Optimizer::generateNoisedTrajectories()
 {
   noise_generator_.setNoisedControls(state_, control_sequence_);
   noise_generator_.generateNextNoises();
-  applyControlConstraints();
   updateStateVelocities(state_);
   integrateStateVelocities(generated_trajectories_, state_);
 }
 
 bool Optimizer::isHolonomic() const {return motion_model_->isHolonomic();}
 
-void Optimizer::applyControlConstraints()
+void Optimizer::applyControlSequenceConstraints()
 {
   auto & s = settings_;
 
   if (isHolonomic()) {
-    state_.cvy = xt::clip(state_.cvy, -s.constraints.vy, s.constraints.vy);
+    control_sequence_.vy = xt::clip(control_sequence_.vy, -s.constraints.vy, s.constraints.vy);
   }
 
-  state_.cvx = xt::clip(state_.cvx, s.constraints.vx_min, s.constraints.vx_max);
-  state_.cwz = xt::clip(state_.cwz, -s.constraints.wz, s.constraints.wz);
-
-  motion_model_->applyConstraints(state_);
```

Comment:
Whoops - that was removed, should readd applied to control_sequence_

Reply:
and probably also need to have the ackermann constraints in the ConstraintCritic...